### PR TITLE
[CL][Tests]: Clean up `FungifyPositions` tests

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Progress towards #5051

## What is the purpose of the change

This PR does a refactor and cleanup of fungification tests to cut down on duplicate code and make assertions simpler/more readable. With these changes, the cases requested in #5051 can be added much more easily.

Specific changes:
* Abstract duplicate code into helpers
* Abstract long checks into helpers to minimize code assertion length
* Remove superfluous error checks
* Remove duplicate assertion
* Move `AddDecCoinsArray` to osmoutils
* Implement and test `CollapseDecCoinsArray` helper in osmoutils

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [x] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/position_test.go [hunk 1]
```diff
@@ -16,6 +16,76 @@ import (
...
+			s.Require().NoError(err)
+			s.Require().False(hasPosition)
+		}
+
+		// Check that the old position has been deleted.
+		_, err := s.App.ConcentratedLiquidityKeeper.GetPosition(s.Ctx, positionId)
+		s.Require().Error(err)
+		s.Require().ErrorAs(err, &types.PositionIdNotFoundError{})
+	}
+}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/position_test.go:38]: It says "on the given accumulators" but it seems like we only check uptime accumulators and not the fee accum. This is fine, but can we just be explicit about only checking the uptime accumulators in the godoc?

File: x/concentrated-liquidity/position_test.go [hunk 2]
```diff
@@ -16,6 +16,76 @@ import (
...
+
+// GetTotalAccruedRewardsByAccumulator returns the total accrued rewards for the given position on each accumulator.
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/position_test.go:40]: Same here, can we be explicit about us only checking uptime accumulator?

**Review by czarcas7ic — APPROVED** [review]
Nice work 👍

File: x/concentrated-liquidity/position_test.go (hunk 1 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/position_test.go]: This was leftover from an older/less general version of the abstraction – fixed now!

**Pull request merged**